app/components/provider_interface/status_box_components/offer_component.rb

```diff
@@ -3,9 +3,13 @@ module StatusBoxComponents
     class OfferComponent < ActionView::Component::Base
       include ViewHelper
       attr_reader :application_choice
+      attr_reader :available_providers, :available_courses, :available_course_options
 
-      def initialize(application_choice:)
-        @application_choice = application_choice
+      def initialize(args)
+        @application_choice = args[:application_choice]
```

Comment:
This seems to duplicate the pattern in `OfferSummaryListComponent` above of passing the collections when only the path is needed. I wonder if we can deduce paths in a common piece of logic before the component is initialized? A helper maybe?

Reply:
Whether to show each link and what the link is, both of which can be controlled by the presence or absence of a link that's passed to the component (and these links would need to be passed through the top-level `StatusBoxComponent` the way things currently are). But then the controller needs to make that decision, which defeats the purpose of the current code structure around status boxes (them being self-contained, the controller doesn't need to worry about them too much).

With the 'change offer' stuff, I think we're pushing the limits of what can be done with standard MVC/Rails, in terms of multi-step wizard flows and steps referring to previous steps etc. The current approach (and action view components, for that matter) is probably influenced by Javascript/MVVM frameworks where a big state entity is made available to components and these just pick what they want. Happy to discuss further, but I think this may be a more general architecture/direction discussion.